src/clients/python/__init__.py

```diff
@@ -1548,10 +1553,11 @@ def get_async_run_results(self, request_id, wait):
         err = c_void_p(_crequest_infer_ctx_get_async_run_results(
             self._ctx, byref(c_is_ready), request_id, wait))
 
-
+        self._lock.acquire()
```

Comment:
Use try...finally.

Why does this acquire need to be here. Can't it be moved down to just before "requested_outputs = ... "

Reply:
Yes, we can allow threads to race for updating _last_request_id. Fixed.